invenio_app_ils/circulation/indexer.py

```diff
@@ -49,3 +57,19 @@ def index(self, loan, arguments=None, **kwargs):
         super().index(loan)
         eta = datetime.utcnow() + current_app.config["ILS_INDEXER_TASK_DELAY"]
         index_referenced_records.apply_async((loan,), eta=eta)
+
+
+def index_item_availability_for_loan(loan_dict):
+    """Indexer hook to modify the loan record dict before indexing."""
```

Comment:
```suggestion
"""Indexer hook to modify the loan record dict before indexing.

     The `available_items_for_loan_count` field is added to the loan only on the search index
     because it is needed for search aggregation/filtering. It is not needed when fetching the
     loan details.
"""
```